src/SchemaDesign/SchemaDesigner.tsx

```diff
@@ -0,0 +1,416 @@
+    const result = Array.from(list)
+    const [removed] = result.splice(startIndex, 1)
+    result.splice(endIndex, 0, removed)
+
+    return result
+  }
+
+  const onQuestionDragEnd = (result) => {
+    if (!result.destination) {
+      return
+    }
+    if (result.destination.index === result.source.index) {
+      return
+    }
+    const stepToAmend = steps.find((step) => step.reference === selectedStep)
+    const updatedQuestions = reorder(stepToAmend.questions, result.source.index, result.destination.index)
+
+    setSteps(steps.map((step) => (step.reference === selectedStep ? { ...step, questions: updatedQuestions } : step)))
+  }
+
+  const deleteStep = (step: string) => {
+    const updatedSteps = steps.filter((stepToRemove) => stepToRemove === step)
+    setSteps(updatedSteps)
+  }
+
+  const getQuestionIcon = (type: string) => {
+    if (type === 'date') {
+      return <CalendarMonthIcon />
+    }
+    if (type === 'boolean') {
+      return <CheckBoxIcon />
+    }
+    return <TextFieldsIcon />
+  }
+
+  return (
+    <Box>
+      <Alert sx={{ mt: 2, mb: 2 }} severity='info'>
+        This page is currently not fully implemented. More advanced schema options will be added in the future.
+      </Alert>
+      <Grid container spacing={2}>
+        <Grid item xs={12} sm={12} md={6}>
+          <Box>
+            <Typography sx={{ mb: 2 }} variant='h4'>
+              Configure Schema
+            </Typography>
+            <Paper
+              sx={{
+                p: 2,
+              }}
+            >
+              <Stack direction='row' spacing={2}>
+                <Tabs value={selectedStep} onChange={handleStepChange} variant='scrollable' scrollButtons='auto'>
+                  {steps.map((step) => (
+                    <Tab
+                      key={step.reference}
+                      value={step.reference}
+                      label={
+                        <span>
+                          {step.title}
+                          <IconButton
+                            size='small'
+                            component='span'
+                            sx={{ ml: 1 }}
+                            onClick={() => {
+                              deleteStep(step.reference)
+                            }}
+                          >
+                            <DeleteIcon />
+                          </IconButton>
+                        </span>
+                      }
+                      onClick={() => setSelectedStep(step.reference)}
+                    />
+                  ))}
+                </Tabs>
+
+                <Button
+                  color='primary'
+                  variant='text'
+                  startIcon={<AddIcon />}
+                  onClick={() => setNewStepDialogOpen(true)}
+                >
+                  Add step
+                </Button>
+              </Stack>
+
+              {steps.map((step) => (
+                <Box key={step.reference}>
```

Comment:
Can be swapped out for a fragment. You can attach a `key` to a fragment like so:
```
import { Fragment } from 'react'

<Fragment key={step.reference}>
  ...
</Fragment>
```